src/make-hid.sh

```diff
@@ -1,25 +1,74 @@
-#!/bin/sh
+#!/bin/bash
+
 MAC=$(echo "$1" | tr a-f A-F)
-DEV=hci0
-CTRL=$(tr a-f A-F </sys/class/bluetooth/$DEV/address)
+HCIDEV=hci0
+PASSEDDEV="$2"
 
 function die() {
   echo "FATAL: $@" 1>&2
   exit 1
 }
 
-[ -z "$CTRL" ] || [ -z "$MAC" ] && die "Call as: $0 keyboard-mac"
+declare -A ahci
+
+ahci=()
+pipe=/tmp/hcitool$$
+trap "rm -f $pipe" EXIT
+
+if [[ ! -p $pipe ]]; then
+    echo  mkfifo $pipe
+fi
+
+hcitool dev > $pipe 2> /dev/null
+while read -t 1 line
+do
+    case $line in
+	hci*)
+	    strs=($line)
+	    echo "	hci[${strs[0]##hci}]: " ${strs[1]}
+	    ahci+=([${strs[0]}]="${strs[1]}")
+	    # echo "hci[${strs[0]}]: "${hci[${strs[0]}]}
+	    ;;
+	*)
+	    echo "$line"
+	    ;;
+    esac
+    
+done < $pipe
+
+if hcitool dev | grep "hci" | wc -l > 1 ; then
+    if [ "$PASSEDDEV" == "" ]; then
+	echo "There are multiple HCI devices."
+	echo "Which HCI device do you want to use?"
+	while read HCIDEV
+	do
+	    if [[ "${ahci[hci$HCIDEV]}" == '' ]] ; then
+		echo "Invalid device [$HCIDEV].  Please enter the correct hci device number?"
+	    else
+		HCIDEV=hci${HCIDEV}
+		break
+	    fi
+	done
+    else
+	HCIDEV=$PASSEDDEV
```

Comment:
I'm not sure why you need to pass the device, can't you just read the address like before, but from the given hci device?

Reply:
On the Raspberry Pi the address want there. Don't know why. You  also need the hci device to pass to the bccmd command. Ideas / suggestions welcome.